drivers/overlay/ostree.go

```diff
@@ -0,0 +1,110 @@
+// +build ostree
+
+package overlay
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/containers/storage/pkg/idtools"
+	"github.com/containers/storage/pkg/system"
+	"github.com/ostreedev/ostree-go/pkg/otbuiltin"
+	"github.com/pkg/errors"
+)
+
+func ostreeSupport() bool {
+	return true
+}
+
+func fixFiles(dir string, usermode bool) error {
+	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if info.Mode()&(os.ModeNamedPipe|os.ModeSocket|os.ModeDevice) != 0 {
+			if err := os.Remove(path); err != nil {
+				return err
+			}
+			if info.IsDir() {
```

Comment:
This looks like it's nested at the wrong level - a directory is unlikely to be a pipe/socket/device.